examples/hosted-wallet-ts/src/pfi/api/exchanges.ts

```diff
@@ -0,0 +1,363 @@
+import { Router } from 'express';
+import axios from 'axios';
+import {
+  BearerDid,
+  Offering,
+  GetExchangeResponseBody,
+  CreateExchangeRequestBody,
+  UpdateExchangeRequestBody,
+  ReplyToRequestBody,
+  Quote,
+  OrderInstructions,
+  OrderStatus,
+  Close,
+  Rfq,
+  Order,
+  Cancel,
+  Message,
+  GetExchangesResponseBody,
+  QuoteData,
+  OrderInstructionsData,
+  OrderStatusData,
+  CloseData,
+  ORDER_STATUS_KIND,
+} from 'tbdex';
+import { OfferingRepository } from '../data/offering-repository';
+
+interface Exchange {
+  rfq?: Message;
+  quote?: Message;
+  order?: Message;
+  orderInstructions?: Message;
+  cancel?: Message;
+  close?: Message;
+  orderStatuses?: Message[];
+}
+
+export default function exchangesRouter(
+  bearerDid: BearerDid,
+  offeringsRepository: OfferingRepository
+) {
+  const router = Router();
+
+  // In-memory storage for exchanges and reply URLs
+  const exchangesToReplyTo = new Map<string, string>();
+  const exchangeIdToExchange = new Map<string, Exchange>();
+
+  // Helper function to introduce delays in async functions
+  const delay = (ms: number) => new Promise((resolve) => setTimeout(resolve, ms));
+
+  /**
+   * GET /exchanges
+   * Retrieves a list of exchange IDs with pagination support
+   */
+  router.get('/', (req, res) => {
```

Comment:
should this be ?

```suggestion
  router.get('/exchanges', (req, res) => {
```

Reply:
oh I see, never mind

Reply:
express prepends that here in the index file - 
```
app.use('/exchanges', exchangesRouter(bearerDid, offeringsRepository));
```